pyccel/parser/semantic.py

```diff
@@ -1738,6 +1746,14 @@ def _ensure_inferred_type_matches_existing(self, class_type, d_var, var, is_auga
                                 self.current_ast_node.col_offset))
 
                 else:
+                    alloc_type = None
+                    if isinstance(getattr(var,'class_type'), (HomogeneousListType, HomogeneousSetType,DictType)):
```

Comment:
Why do you use getattr instead of:
```suggestion
                    if isinstance(var.class_type, (HomogeneousListType, HomogeneousSetType,DictType)):
```
?

Reply:
That's what i wanted to use at the beginning but I found out that not all variables have that attribute.

Reply:
All semantic objects inheriting from `TypedAstNode` should have this property. So maybe you found a bug? Without a default argument this function is doing the same as my proposition though:

![image](https://github.com/user-attachments/assets/e500bbc5-a2e9-4f9c-8042-b11c4c9e6f9c)

So I'm not sure why this would fix anything as is. Maybe the problem was there earlier?